Scripts/symbolic_hermitians/generate_code.py

```diff
@@ -362,7 +362,12 @@ def sgn(t,var):
         for fii in fdlist:
             code.append("sumP += " + fii + ";")
         code.append("error = sumP-1.0;")
-        code.append("if( std::abs(error) > 100.*parms->maxError) amrex::Abort();")
+        code.append('if( std::abs(error) > 100.*parms->maxError) {')
+        code.append("std::ostringstream Convert;")
+        code.append('Convert << "Matrix trace (SumP) is not equal to 1, trace error exceeds 100*maxError: " << std::abs(error) << " > " << 100.*parms->maxError;')
+        code.append("std::string Trace_Error = Convert.str();")
+        code.append('amrex::Error(Trace_Error);')
```

Comment:
This looks good! I think I would have personally just written amrex::Error(Convert.str()); to avoid defining more variables, but that is just a style preference.

Reply:
Oh yeah, you're right. I can change it to that if you'd like, though I guess it doesn't make a big difference.